public/app/features/profile/UserProfileEditPage.tsx

```diff
@@ -55,19 +63,92 @@ export function UserProfileEditPage({
   changeUserOrg,
   updateUserProfile,
 }: Props) {
+  const [queryParams, updateQueryParams] = useQueryParams();
+  const tabQueryParam = queryParams[TAB_QUERY_PARAM];
+  const [activeTab, setActiveTab] = useState<string>(
+    (typeof tabQueryParam === 'string' ? tabQueryParam : CORE_SETTINGS_TAB).toLowerCase()
+  );
+
   useMount(() => initUserProfilePage());
 
+  const extensionComponents = useMemo(() => {
+    const { extensions } = getPluginComponentExtensions({
+      extensionPointId: PluginExtensionPoints.UserProfileTab,
+      context: {},
+    });
+
+    return extensions;
+  }, []);
+
+  const groupedExtensionComponents = extensionComponents.reduce<Record<string, PluginExtensionComponent[]>>(
+    (acc, extension) => {
+      const { title } = extension;
+      if (acc[title]) {
+        acc[title].push(extension);
+      } else {
+        acc[title] = [extension];
+      }
+      return acc;
+    },
+    {}
+  );
+
+  const showTabs = extensionComponents.length > 0;
+  const tabs = [CORE_SETTINGS_TAB, ...keys(groupedExtensionComponents).map((title) => title)];
+
+  const UserProfile = () => (
+    <VerticalGroup spacing="md">
+      <UserProfileEditForm updateProfile={updateUserProfile} isSavingUser={isUpdating} user={user} />
+      <SharedPreferences resourceUri="user" preferenceType="user" />
+      <UserTeams isLoading={teamsAreLoading} teams={teams} />
+      <UserOrganizations isLoading={orgsAreLoading} setUserOrg={changeUserOrg} orgs={orgs} user={user} />
+      <UserSessions isLoading={sessionsAreLoading} revokeUserSession={revokeUserSession} sessions={sessions} />
+    </VerticalGroup>
+  );
+
+  const UserProfileWithTabs = () => (
+    <div data-testid={selectors.components.UserProfile.extensionPointTabs}>
+      <VerticalGroup spacing="md">
+        <TabsBar>
+          {tabs.map((tabTitle) => {
+            const queryParamCompatibleTabTitle = tabTitle.toLowerCase();
+
+            return (
+              <Tab
+                key={tabTitle}
+                label={tabTitle}
+                active={activeTab === queryParamCompatibleTabTitle}
+                onChangeTab={() => {
+                  setActiveTab(queryParamCompatibleTabTitle);
+                  updateQueryParams({ [TAB_QUERY_PARAM]: queryParamCompatibleTabTitle });
+                }}
+                data-testid={selectors.components.UserProfile.extensionPointTab(tabTitle)}
+              />
+            );
+          })}
```

Comment:
with the new `tabs` structure, you can use `id` pretty much everywhere and only use `title` for the `label`